### DISCUSSION TITLE
Filename extension association editing

### DISCUSSION THREAD

**Discussion by jwoehr** [codefori/vscode-ibmi#2310] [General]
I see the `files.associations` object in the `settings.json` file for Code for i but I'm not clear how to use the Gui to change these associations other that directly editing the `settings.json` file. E.g., the extension `*.d2w` is assigned to `netdata` but the extension `*.ndm` should also be. Am I missing a navigation path?

**Comment by SanjulaGanepola** [comment]
`files.associations` is actually a VS Code setting and not a Code4i specific setting. One option to change these associations is to open a file with the extension you want changed and then use the `Change Language Mode` command with the `Configure File Association...` option:

![image](https://github.com/user-attachments/assets/ca7cdf9e-a7b3-46d7-88e3-44f8fbc70a4b)

It can also be easily opened from the status bar in the bottom right:

![image](https://github.com/user-attachments/assets/928662bc-fc18-4150-861c-c3e5ae9ffd26)

Another option to edit and view all associations is to use the VS Code Settings GUI:

![image](https://github.com/user-attachments/assets/af13461a-8ccc-47a9-a714-f652b89d1885)

**Comment by jwoehr** [comment]
Ah, thank you @SanjulaGanepola ... that works for me.
